ui/app/src/components/FormsEngine/FormsEngine.tsx

```diff
@@ -0,0 +1,2464 @@
+    } else {
+      doClose();
+    }
+  };
+  // region const tableOfContents = (...)
+  const useCollapsedToC = isLargeContainer ? collapsedToC : true;
+  const tableOfContents = (
+    <TableOfContents
+      handleSectionExpandedChange={handleSectionExpandedChange}
+      atoms={stableFormContext.atoms}
+      fieldsToRender={fieldsToRender}
+      containerRef={containerRef}
+      contentTypeFields={contentTypeFields}
+      contentTypeSections={contentTypeSections}
+      sectionExpandedState={sectionExpandedState}
+      setOpenDrawerSidebar={setOpenDrawerSidebar}
+    />
+  );
+  // endregion
+
+  const currentStackedFormProps = hasStackedForms ? formsStackData[formsStackData.length - 1].props : null;
+  let stackedFormKey = undefined;
+  if (hasStackedForms) {
+    if (currentStackedFormProps.update) {
+      stackedFormKey = `${currentStackedFormProps.update.path}_${currentStackedFormProps.update.modelId ?? ''}_${stackFormCount}`;
+    } else if (currentStackedFormProps.create) {
+      stackedFormKey = `${currentStackedFormProps.create.path}_${currentStackedFormProps.create.contentTypeId}_${stackFormCount}`;
+    } else if (currentStackedFormProps.repeat) {
+      stackedFormKey = `${currentStackedFormProps.repeat.fieldId}_${stackFormCount}`;
+    }
+  }
+
+  const disableSave = isSubmitting || (affectedPackages && !acceptedWorkflowCancellation);
+  const handleSave: ButtonProps['onClick'] = (e?: React.MouseEvent<HTMLButtonElement>) => {
+    // TODO: VALIDATIONS
+    const values = extractValueAtoms(store, stableFormContext.atoms.valueByFieldId);
+    const closeInstruction = () =>
+      setTimeout(() => {
+        // Controls like the Item Selector use the onSave so that once the form saves, the control of the underlying form gets updated.
+        // Executing the onClose without the timeout, causes values set at the control prior to closing to get lost somehow.
+        // Putting the timeout at the control works too, but prefer to simply it for controls and absorb the complexity here.
+        (isStackedForm ? onCloseProp : enhancedDialogOnClose)?.(e, null);
+      });
+    // Repeat handled here. Execution ends inside if statement.
+    if (isRepeatMode) {
+      setHasPendingChanges(false);
+      const instructions = onSave?.({ values, versionComment });
+      if (instructions?.close) closeInstruction();
+      return;
+    }
+    const date = new Date().toISOString();
+    // Put system properties in xml before creating the XML
+    values[XmlKeys.fileName] = values[XmlKeys.fileName] ?? contentObject[XmlKeys.fileName];
+    values[XmlKeys.folderName] = values[XmlKeys.folderName] ?? contentObject[XmlKeys.folderName];
+    if (
+      String(values[XmlKeys.internalName]).trim() === '' ||
+      String(values[XmlKeys.fileName]).trim() === '' ||
+      String(values[XmlKeys.folderName]).trim() === ''
+    ) {
+      return showAlert({ dispatch, message: 'You need a page url and internal name at a minimum to save content.' });
```

Comment:
URL, not url. Can you fix acronyms throughout? Should that be a separate ticket?